components/message.php

```diff
@@ -0,0 +1,504 @@
+			'max'          => $r['count'],
+			'search_terms' => $r['search'],
+		) );
+
+		if ( ! $inbox->has_threads() ) {
+			WP_CLI::error( 'No messages found.' );
+		}
+
+		$messages = $inbox->threads[0]->messages;
+
+		if ( 'ids' === $formatter->format ) {
+			echo implode( ' ', wp_list_pluck( $messages, 'id' ) ); // WPCS: XSS ok.
+		} elseif ( 'count' === $formatter->format ) {
+			$formatter->display_items( $messages );
+		} else {
+			$formatter->display_items( $messages );
+		}
+	}
+
+	/**
+	 * Generate random messages.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--thread-id=<thread-id>]
+	 * : Thread ID to generate messages against.
+	 * ---
+	 * default: false
+	 * ---
+	 *
+	 * [--count=<number>]
+	 * : How many messages to generate.
+	 * ---
+	 * default: 20
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp message generate --thread-id=6465 --count=10
+	 *     $ wp bp message generate --count=100
+	 */
+	public function generate( $args, $assoc_args ) {
+		$notify = \WP_CLI\Utils\make_progress_bar( 'Generating messages', $assoc_args['count'] );
+
+		for ( $i = 0; $i < $assoc_args['count']; $i++ ) {
+			$this->create( array(), array(
+				'from'      => $this->get_random_user_id(),
+				'to'        => $this->get_random_user_id(),
+				'subject'   => sprintf( 'Message Subject - #%d', $i ),
+				'thread-id' => $assoc_args['thread-id'],
+				'silent'    => true,
+			) );
+
+			$notify->tick();
+		}
+
+		$notify->finish();
+	}
+
+	/**
+	 * Star a message.
+	 *
+	 * ## OPTIONS
+	 *
+	 * --message-id=<message-id>
+	 * : Message ID to star.
+	 *
+	 * --user-id=<user>
+	 * : User that is starring the message. Accepts either a user_login or a numeric ID.
+	 *
+	 * ## EXAMPLE
+	 *
+	 *     $ wp bp message star --message-id=3543 --user-id=user_login
+	 *     Success: Message was successfully starred.
+	 */
+	public function star( $args, $assoc_args ) {
+		$user = $this->get_user_id_from_identifier( $assoc_args['user-id'] );
+		if ( ! $user ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		$user_id = $user->ID;
+		$msg_id  = (int) $assoc_args['message-id'];
+
+		if ( bp_messages_is_message_starred( $msg_id, $user_id ) ) {
+			WP_CLI::error( 'The message is already starred.' );
+		}
+
+		$star_args = array(
+			'action'     => 'star',
+			'message_id' => $msg_id,
+			'user_id'    => $user_id,
+		);
+
+		if ( bp_messages_star_set_action( $star_args ) ) {
+			WP_CLI::success( 'Message was successfully starred.' );
+		} else {
+			WP_CLI::error( 'Message was not starred.' );
+		}
+	}
+
+	/**
+	 * Unstar a thread.
+	 *
+	 * ## OPTIONS
+	 *
+	 * --thread-id=<thread-id>
+	 * : Thread ID to unstar.
+	 *
+	 * --user-id=<user>
+	 * : User that is unstarring the thread. Accepts either a user_login or a numeric ID.
+	 *
+	 * ## EXAMPLE
+	 *
+	 *     $ wp bp message unstar --thread-id=212 --user-id=another_user_login
+	 *     Success: Message was successfully unstarred.
+	 */
+	public function unstar( $args, $assoc_args ) {
```

Comment:
`star` stars a specific message, `unstar` unstars a whole thread. For consistency, `unstar` should unstar a single message. Unstarring all messages in a thread should either be `unstar-thread`, or `wp bp message unstar --thread-id` (how it currently is, but allowing `wp bp message unstar --message-id=123`)